modules/auth/auth.go

```diff
@@ -305,6 +305,10 @@ func validate(errs binding.Errors, data map[string]interface{}, f Form, l macaro
 	}
 
 	data["HasError"] = true
+	// If the field with name errs[0].FieldNames[0] is not found in form
+	// somehow, some code later on will panic on Data["ErrorMsg"].(string).
+	// So initialize it to some default.
+	data["ErrorMsg"] = l.Tr("form.unknown_error")
```

Comment:
This change is not really needed for my pull request. Before I made changes below, the code was unable to find the field that failed validation inside embedded anonymous struct, and I got panic due ErrorMsg being not set. I decided to add this so the code won't crash in similar cases (which should not happen unless there's a bug).

In my opinion, the entire function is somewhat sloppy and needs rewriting. I only modified it as much as I needed, and didn't try to refactor it.